File: app/src/main/java/com/jingxi/smartlife/pad/mvp/home/view/MessageDialog.java
package com.jingxi.smartlife.pad.mvp.home.view;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.jingxi.smartlife.pad.R;
import com.wujia.businesslib.data.MsgDto;

/**
 * author ：shenbingkai@163.com
 * date ：2019-02-17
 * description ：
 */
public class MessageDialog extends Dialog {

    public interface MsgReadCallBack {
        void updateMsgReadStatus(MsgDto.ContentBean msg);
    }

    MsgReadCallBack msgReadCallBack;

    public MessageDialog(@NonNull Context context, final MsgDto.ContentBean message) {
        super(context);
//        mContext = context;
        setContentView(R.layout.dialog_msg_layout);
        TextView btnCall = findViewById(R.id.btn1);
        TextView btnKnow = findViewById(R.id.btn2);

        TextView tvType = findViewById(R.id.tv1);
        TextView tvTitle = findViewById(R.id.tv2);
        TextView tvTime = findViewById(R.id.tv3);
        TextView tvDesc = findViewById(R.id.tv4);
        ImageView icon = findViewById(R.id.img1);

        if (message.getType().equals(MsgDto.TYPE_NOTIFY)) {
//            icon.setImageResource(R.mipmap.ic_msg_label_neighbour);
            icon.setImageResource(R.mipmap.ic_msg_label_neighbour2);
        } else if (message.getType().equals(MsgDto.TYPE_PROPERTY)) {
//            icon.setImageResource(R.mipmap.ic_msg_label_serve);
            icon.setImageResource(R.mipmap.ic_msg_label_serve2);
        } else if (message.getType().equals(MsgDto.TYPE_SYSTEM)) {
            icon.setImageResource(R.mipmap.icon_msg_label_system);
        }

        tvType.setText(MsgDto.getTypeText(message));
        tvTitle.setText(message.getTitle());
//        tvTime.setText(DateUtil.formatMsgDate(message.createDate));
        tvTime.setText(message.getCreateDate());
        tvDesc.setText(message.getContent());

        if (message.getIsRead() == MsgDto.STATUS_UNREAD) {//未读
            btnKnow.setBackgroundResource(R.drawable.btn_rect_accent_select);
        } else {
            btnKnow.setBackgroundResource(R.drawable.btn_rect_no_can);
        }
        btnKnow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (message.getIsRead() == MsgDto.STATUS_UNREAD) {//未读
                    v.setBackgroundResource(R.drawable.btn_rect_no_can);

                    if (null != msgReadCallBack) {
                        msgReadCallBack.updateMsgReadStatus(message);
                    }
                }
                dismiss();
            }
        });

    }

    public MessageDialog setListener(MsgReadCallBack listener) {
        this.msgReadCallBack = listener;
        return this;
    }

//    @Override
//    public void onClick(View v) {
//        if (v.getId() == R.id.btn1) {
//            dismiss();
//            new CallDialog(mContext).show();
//        } else if (v.getId() == R.id.btn2) {
//            dismiss();
//        }
//    }
}

File: app/src/main/java/com/jingxi/smartlife/pad/mvp/home/view/MessageDialog.kt
package com.jingxi.smartlife.pad.mvp.home.view

import android.app.Dialog
import android.content.Context
import com.jingxi.smartlife.pad.R
import com.wujia.businesslib.data.MsgDto
import kotlinx.android.synthetic.main.dialog_msg_layout.*

/**
 * author ：shenbingkai@163.com
 * date ：2019-02-17
 * description ：
 */
class MessageDialog(context: Context, message: MsgDto.ContentBean) : Dialog(context) {

    private var msgReadCallBack: MsgReadCallBack? = null
    private var listner: ((msg: MsgDto.ContentBean)->Unit)? = null

    interface MsgReadCallBack {
        fun updateMsgReadStatus(msg: MsgDto.ContentBean)
    }

    init {
        setContentView(R.layout.dialog_msg_layout)//未读

        //未读
        when (message.type) {
            MsgDto.TYPE_NOTIFY -> img1.setImageResource(R.mipmap.ic_msg_label_neighbour2)
            MsgDto.TYPE_PROPERTY -> img1.setImageResource(R.mipmap.ic_msg_label_serve2)
            MsgDto.TYPE_SYSTEM -> img1.setImageResource(R.mipmap.icon_msg_label_system)
            MsgDto.TYPE_ORDER -> img1.setImageResource(R.mipmap.icon_msg_label_system)
            MsgDto.TYPE_GOV -> img1.setImageResource(R.mipmap.icon_msg_label_system)
        }

        tvType.text = MsgDto.getTypeText(message)
        tvTitle.text = message.title
        //        tvTime.setText(DateUtil.formatMsgDate(message.createDate));
        //        tvTime.setText(DateUtil.formatMsgDate(message.createDate));
        tvTime.text = message.createDate
        tvDesc.text = message.content

        if (message.isRead == MsgDto.STATUS_UNREAD) {//未读
            btnKnow.setBackgroundResource(R.drawable.btn_rect_accent_select)
        } else {
            btnKnow.setBackgroundResource(R.drawable.btn_rect_no_can)
        }
        btnKnow.setOnClickListener { v ->
            if (message.isRead == MsgDto.STATUS_UNREAD) {//未读
                v.setBackgroundResource(R.drawable.btn_rect_no_can)
                msgReadCallBack?.updateMsgReadStatus(message)
                listner?.invoke(message)
            }
            dismiss()
        }

    }

    fun setListener(listener: ((msg: MsgDto.ContentBean)->Unit)?): MessageDialog {
        this.listner = listener
        return this
    }
}
